### PR TITLE
Pending tx get

### PR DESCRIPTION
**Pull request by nuevoalex**

### CODE REVIEW

File: packages/augur-sdk/src/Augur.ts [hunk 1]
```diff
@@ -101,4 +101,16 @@ export class Augur<TProvider extends Provider = Provider> {
   public getOrders():ContractInterfaces.Orders {
     return new ContractInterfaces.Orders(this.dependencies, this.addresses.Orders);
   }
+
+  public registerTransactionStatusCallback(key: string, callback: TransactionStatusCallback): void {
```
Comment (bthaile) [inline packages/augur-sdk/src/Augur.ts:105]: at the SDK level couldn't `key` be typed? is it `event name` or transaction type? I think we'd benefit with some typing here if I understand how it's going to be used by the UI.

Comment (nuevoalex) [reply packages/augur-sdk/src/Augur.ts:105]: Its arbitrary. its the name of the generic tx status update handler as provided by the consumer

Comment (bthaile) [reply packages/augur-sdk/src/Augur.ts:105]: I see in the test "test" is used what would be an example the UI or other consumer would use?

Comment (justinbarry) [reply packages/augur-sdk/src/Augur.ts:105]: The key exists to give the consumer the ability to deregister the callback. I suppose we could return a closure the  does the same thing but that is more stuff to manage.

Comment (nuevoalex) [reply packages/augur-sdk/src/Augur.ts:105]: Yea. In practice it would be something like "Notifications" for a handler specific to notifications.

Comment (bthaile) [reply packages/augur-sdk/src/Augur.ts:105]: gotcha, that makes sense. so we'd only have one handler to manage tx in UI.

Comment (justinbarry) [reply packages/augur-sdk/src/Augur.ts:105]: If the same key is used twice that would kill the previous callback. Is that what we want here?

**Review by bthaile — APPROVED** [review]
lgtm

File: packages/contract-dependencies-ethers/src/ContractDependenciesEthers.ts [hunk 2]
```diff
@@ -83,13 +117,39 @@ export class ContractDependenciesEthers implements Dependencies<BigNumber> {
...
         if (!this.signer) throw new Error("Attempting to sign a transaction while not providing a signer");
         // TODO: figure out a way to propagate a warning up to the user in this scenario, we don't currently have a mechanism for error propagation, so will require infrastructure work
         // TODO: https://github.com/ethers-io/ethers.js/issues/321
         const tx = this.transactionToEthersTransaction(transaction);
         delete tx.from;
-        const receipt = await (await this.signer.sendTransaction(tx)).wait();
+        const txMetadataKey = `0x${transaction.data.substring(10)}`;
+        const txMetadata = this.transactionDataMetaData[txMetadataKey];
+        this.onTransactionStatusChanged(txMetadata, TransactionStatus.AWAITING_SIGNING);
+        const response = await this.signer.sendTransaction(tx);
```
Comment (justinbarry) [inline packages/contract-dependencies-ethers/src/ContractDependenciesEthers.ts]: If this throws an exception we will have excess txMetadata hanging around. Would be good to handle if we could.

File: packages/augur-sdk/src/Augur.ts (hunk 1 above)
Comment (nuevoalex) [reply packages/augur-sdk/src/Augur.ts:105]: I think that's fine personally. Alternative is to throw an error which I think is obnoxious.

**Review by justinbarry — CHANGES_REQUESTED** [review]
Looks *really* good. Just one feedback item.

File: packages/contract-dependencies-ethers/src/ContractDependenciesEthers.ts (hunk 2 above)
Comment (nuevoalex) [reply packages/contract-dependencies-ethers/src/ContractDependenciesEthers.ts]: will do

**Review by justinbarry — APPROVED** [review]

**Pull request merged**